src/protocol/tx-validity.md

```diff
@@ -112,48 +121,127 @@ def sum_predicate_gas_used(tx) -> int:
             total += input.predicateGasUsed
     return total
 
-"""
-Returns any minted amounts by the transaction
-"""
+    
+def sum_tx_bytes_gas_used(tx) -> int:
+    """
+    Computes the intrinsic gas cost of a transaction based on size in bytes
+    """
+    size(tx) * GAS_PER_BYTE
+
+
 def minted(tx, asset_id) -> int:
+    """
+    Returns any minted amounts by the transaction
+    """
     if tx.type != TransactionType.Mint or asset_id != tx.mint_asset_id:
         return 0
     return tx.mint_amount
 
+
 def sum_outputs(tx, asset_id) -> int:
     total: int = 0
     for output in tx.outputs:
         if output.type == OutputType.Coin and output.asset_id == asset_id:
             total += output.amount
     return total
 
+
+def sum_input_intrinsic_fees(tx) -> int:
```

Comment:
Returning fees instead of gas was an idea I was toying around with to make it easier to transition to multi-dimensional pricing in the future.  However, feel free to change it to more closely match the current implementation.

Reply:
Updated